.github/workflows/deploy-staging.yml

```
@@ -0,0 +1,80 @@
+      - name: Install deps (frontend)
+        working-directory: ./frontend
+        run: pnpm i --frozen-lockfile
+      - name: Build frontend
+        working-directory: ./frontend
+        run: pnpm build
+
+      # -------- Deploy frontend (rsync) --------
+      - name: Deploy frontend dist
+        env:
+          SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2086:info:2:88: Double quote to prevent globbing and word splitting

(shellcheck)